File: py/solns/validParentheses/validParentheses.py

class Solution:
    @staticmethod
    def naive(s):
        stack = []
        d = {"(":")","{":"}","[":"]"}
        for tmps in s:
            if tmps in d:
                stack.append(d[tmps])
            if tmps in d.values():
                if len(stack)==0: return False
                tmp = stack.pop()
                if tmp!=tmps:
                    return False
        return len(stack)==0

File: cs/sample/Sample/ValidParentheses/ValidParentheses.cs
using System;
using System.Collections.Generic;

namespace Sample.ValidParentheses{
    public class validParentheses{
        public static bool naive(string s){
            Stack<char> st = new Stack<char>();
            Dictionary<char,char> d = new Dictionary<char,char>(){
                {'(',')'},{'[',']'},{'{','}'}};
            HashSet<char> close = new HashSet<char>(){')','}',']'};
            foreach (char c in s){
                if (d.ContainsKey(c))
                    st.Push(d[c]);
                if (close.Contains(c)){
                    if (st.Count==0) return false;
                    char tmp = st.Pop();
                    if (tmp!=c) return false;
                }
            }
            return st.Count==0;
        }
    }
}
